### PR TITLE
Bump rails from 5.2.3 to 6.0.2.1

### PR DESCRIPTION
**Pull request by MuriloDalRi**
Bumps [rails](https://github.com/rails/rails) from 5.2.3 to 6.0.2.1.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a href="https://github.com/rails/rails/releases">rails's releases</a>.</em></p>
<blockquote>
<h2>6.0.2.1</h2>
<h2>Active Support</h2>
<ul>
<li>No changes.</li>
</ul>
<h2>Active Model</h2>
<ul>
<li>No changes.</li>
</ul>
<h2>Active Record</h2>
<ul>
<li>No changes.</li>
</ul>
<h2>Action View</h2>
<ul>
<li>No changes.</li>
</ul>
<h2>Action Pack</h2>
<ul>
<li>
<p>Fix possible information leak / session hijacking vulnerability.</p>
<p>The <code>ActionDispatch::Session::MemcacheStore</code> is still vulnerable given it requires the
gem dalli to be updated as well.</p>
<p>CVE-2019-16782.</p>
</li>
</ul>
<h2>Active Job</h2>
<ul>
<li>No changes.</li>
</ul>
<h2>Action Mailer</h2>
<ul>
<li>No changes.</li>
</ul>
<h2>Action Cable</h2>
</tr></table> ... (truncated)
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a href="https://github.com/rails/rails/commit/f33d52c95217212cbacc8d5e44b5a8e3cdc6f5b3"><code>f33d52c</code></a> Preparing for 6.0.2.1 release</li>
<li><a href="https://github.com/rails/rails/commit/ad2da1ad82742354a73cf30be658574d9bd567f0"><code>ad2da1a</code></a> Fix possible information leak / session hijacking vulnerability.</li>
<li><a href="https://github.com/rails/rails/commit/f675cb30ce813a99b52b139a93e048330922fd9a"><code>f675cb3</code></a> Preparing for 6.0.2 release</li>
<li><a href="https://github.com/rails/rails/commit/688e5230e6566a4aa2602204fc93014947fe2b3e"><code>688e523</code></a> Fix release task again</li>
<li><a href="https://github.com/rails/rails/commit/63107e9914c893336f7612c2cd17a24474b6a6d6"><code>63107e9</code></a> Preparing for 6.0.2.rc2 release</li>
<li><a href="https://github.com/rails/rails/commit/ec7721c0d95fc4da1bda9f50feb2efae4ab2a55a"><code>ec7721c</code></a> Revert &quot;Merge pull request <a href="https://github-redirect.dependabot.com/rails/rails/issues/37504">#37504</a> from utilum/no_implicit_conversion_of_nil&quot;</li>
<li><a href="https://github.com/rails/rails/commit/296cb7f2f447f05c92a9f2abddd2b949cde88a9b"><code>296cb7f</code></a> unlinks Ruby on Rails Tutorial [skip ci]</li>
<li><a href="https://github.com/rails/rails/commit/6629ed7a5e2aaeb21a5a25ceeeb1e98aa77f1598"><code>6629ed7</code></a> Controller can be symbols as well</li>
<li><a href="https://github.com/rails/rails/commit/c3135a4d4050496355d489b0deb84db54b8f0553"><code>c3135a4</code></a> Revert &quot;Merge pull request <a href="https://github-redirect.dependabot.com/rails/rails/issues/37849">#37849</a> from kamipo/fix_since_and_ago&quot;</li>
<li><a href="https://github.com/rails/rails/commit/fcfe6931d3f14dfd8a635cb1c5448ba1348853bb"><code>fcfe693</code></a> Revert &quot;Merge pull request <a href="https://github-redirect.dependabot.com/rails/rails/issues/37839">#37839</a> from ttanimichi/modify-inspect-of-activesup...</li>
<li>Additional commits viewable in <a href="https://github.com/rails/rails/compare/v5.2.3...v6.0.2.1">compare view</a></li>
</ul>
</details>
<br />

[Trello](https://trello.com/c/zHgPEjRD/1758-5-upgrade-email-alert-api-to-rails-6)

[Dependabot compatibility score](https://dependabot.com/compatibility-score.html?dependency-name=rails&package-manager=bundler&previous-version=5.2.3&new-version=6.0.2.1)

### CODE REVIEW

**Comment by thomasleese** [comment]
We should also [run `app:update`](https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#the-update-task) to make sure we have the latest files in `config/`.

**Comment by MuriloDalRi** [comment]
> We should also [run `app:update`](https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#the-update-task) to make sure we have the latest files in `config/`.

I forgot 😢

**Comment by MuriloDalRi** [comment]
@thomasleese I wasn't super sure about some of the changes so I kept the old version in some places. Does it look ok?

File: app/workers/process_content_change_and_generate_emails_worker.rb [hunk 1]
```diff
@@ -5,8 +5,10 @@ class ProcessContentChangeAndGenerateEmailsWorker < ProcessAndGenerateEmailsWork
 
   def perform(content_change_id)
     content_change = ContentChange.find(content_change_id)
+
     return if content_change.processed?
 
+
```
Comment (cbaines) [inline app/workers/process_content_change_and_generate_emails_worker.rb]: Maybe get rid of these changes (or move them to a commit that says why blank lines are being added).

File: spec/workers/process_content_change_and_generate_emails_worker_spec.rb [hunk 2]
```diff
@@ -209,6 +210,9 @@
 
       expect(Email.count).to eq(100)
       expect(Email.all.pluck(:address).uniq.count).to eq(100)
+      DatabaseCleaner.allow_remote_database_url = true
+      DatabaseCleaner.strategy = :truncation
+      DatabaseCleaner.clean
```
Comment (cbaines) [inline spec/workers/process_content_change_and_generate_emails_worker_spec.rb]: We need this to happen after the test, regardless of what happens during the test. While having this code at the end was sufficient for testing, if the test crashes before the end, this cleanup won't happen.

Is there some way of telling RSpec to do this after this test?

**Comment by thomasleese** [comment]
Looks good to me, since there are some of my commits in here I think it's better for someone else to approve though.

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -30,6 +30,17 @@
   config.after type: :request do
     logout
   end
+
+  config.around(:each, testing_transactions: true) do |example|
+    DatabaseCleaner.allow_remote_database_url = true
+    self.use_transactional_tests = false
+    DatabaseCleaner.strategy = :truncation
+    DatabaseCleaner.clean
```
Comment (cbaines) [inline spec/rails_helper.rb]: This `DatabaseCleaner.clean` shouldn't be necessary, as the test should start with the database in a clean state. Is it necessary?

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -30,6 +30,17 @@
...
+    example.run
+    DatabaseCleaner.clean
+    DatabaseCleaner.strategy = :transaction
```
Comment (cbaines) [inline spec/rails_helper.rb]: `DatabaseCleaner.strategy = :transaction` shouldn't be necessary here, as DatabaseCleaner isn't being used for other tests, right?

File: app/workers/process_content_change_and_generate_emails_worker.rb (hunk 1 above)
Comment (cbaines) [reply app/workers/process_content_change_and_generate_emails_worker.rb]: I can see that you've removed these lines in a subsequent commit. That's better, but still not as good as not adding them in the first place. I'll forgive you though as you're not using Emacs yet.

File: spec/rails_helper.rb (hunk 4 above)
Comment (MuriloDalRi) [reply spec/rails_helper.rb]: You're right, I'll remove that

**Comment by MuriloDalRi** [comment]
@cbaines I think I made all the changes now

**Comment by cbaines** [comment]
> @cbaines I think I made all the changes now

Great, this looks good to me :+1:

**Pull request merged**